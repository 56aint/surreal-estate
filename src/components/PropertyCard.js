import React from "react";
import PropTypes from "prop-types";
import { FaSave, FaArchive } from "react-icons/fa";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faBath, faBed, faEnvelope } from "@fortawesome/free-solid-svg-icons";
// import "../styles/PropertyCard.css";

const PropertyCard = ({
  _id,
  title,
  type,
  bathrooms,
  bedrooms,
  price,
  city,
  email,
  userID,
  onSaveProperty,
}) => {
  return (
    <div className="property-card">
      <div>
        <span className="property-card_title" data-testid="p_title-id">
          {title}
        </span>
      </div>

      <div>
        <span className="property-card_type" data-testid="p_type-id">
          {type}
        </span>
      </div>

      <div>
        <span className="property-card_bathrooms" data-testid="p_bathrooms-id">
          <FontAwesomeIcon icon={faBed} />
          {bathrooms}
        </span>
      </div>

      <div>
        <span className="property-card_bedrooms" data-testid="p_bedrooms-id">
          <FontAwesomeIcon icon={faBath} />
          {bedrooms}
        </span>
      </div>

      <div>
        <span className="property-card_price" data-testid="p_price-id">
          {price}
        </span>
      </div>

      <div>
        <span className="property-card_city" data-testid="p_city-id">
          {city}
        </span>
      </div>

      <div>
        <a href={`mailto:${email}`}>
          <span className="property-card_email" data-testid="p_email-id">
            <FontAwesomeIcon icon={faEnvelope} />
            Email
          </span>
        </a>
      </div>
      {userID && (
        <button
          href="#"
          className="save-button"
          type="button"
          onClick={() => onSaveProperty(_id)}
        >
          <FaArchive className="save-sign" />
          Save
          <FaSave className="save-sign" />
        </button>
      )}
    </div>
  );
};

PropertyCard.propTypes = {
  title: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  bathrooms: PropTypes.string.isRequired,
  bedrooms: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  city: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
  userID: PropTypes.string.isRequired,
  onSaveProperty: PropTypes.func.isRequired,
  _id: PropTypes.string.isRequired,
};

export default PropertyCard;
