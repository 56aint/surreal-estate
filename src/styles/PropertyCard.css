.property-card {
  display: flex;
  flex-direction: column;
  
  justify-content: center;
  height: 150px;
  padding: 10px;
  margin: auto;
  font-family: "Open Sans"; 
  border: 3px solid #141313;
  width: 50%;
  
}
.property-card_price::before {
  content: "\00a3";
  padding: 1px;
}
.property-card_email {
  color: aliceblue;
  background-color: rgb(151, 3, 3);
  width: 100%;
  font-size: medium;
}
.property-card_title {
  font-weight: bold;
}
.save-button {
  width: 85px;
  align-self: left;
  
  margin-right: 10em ;
  
  font-weight: bolder;
  text-align: center;
  background-color: rgb(20, 19, 19);  
  border-radius: 5px;
  color: white;
  padding: .1em;
  text-decoration: none;
}
.save-button:focus,
.save-button:hover {
  background-color: rgb(136, 2, 2);
  color: White;
}

