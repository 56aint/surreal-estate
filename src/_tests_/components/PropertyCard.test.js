import React from "react";
import { render } from "@testing-library/react";
import PropertyCard from "../../components/PropertyCard";

describe("PropertyCard", () => {
  it("renders the correctly", () => {
    const { asFragment } = render(
      <PropertyCard
        title="1 bedroom house"
        type="Detached"
        bathrooms="3"
        bedrooms="1"
        price="450000"
        city="Brighton"
        email="no@no.email.com"
      />
    );
    expect(asFragment).toMatchSnapshot();
  });

  /* xit("renders the correct props", () => {
    const { getByTestId } = render(
      <PropertyCard
        title=""
        type=""
        bathrooms=""
        bedrooms=""
        price=""
        city=""
        email=""
      />
    );
    expect(getByTestId("p_title-id")).toHaveClass("date");
    expect(getByTestId("p_type-id")).toHaveClass("type");
    expect(getByTestId("p_bathrooms-id")).toHaveClass("bathrooms");
    expect(getByTestId("p_bedrooms-id")).toHaveClass("bedrooms");
    expect(getByTestId("p_price-id")).toHaveClass("price");
    expect(getByTestId("p_email-id")).toHaveClass("email");
  }); */
});
