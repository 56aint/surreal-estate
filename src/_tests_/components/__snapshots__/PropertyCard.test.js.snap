// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard renders correctly 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    <div>
      <span
        class="property-card_title"
        data-testid="p_title-id"
      >
        1 bedroom house
      </span>
    </div>
    <div>
      <span
        class="property-card_type"
        data-testid="p_type-id"
      >
        Detached
      </span>
    </div>
    <div>
      <span
        class="property-card_bathrooms"
        data-testid="p_bathrooms-id"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-bed fa-w-20 "
          data-icon="bed"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M176 256c44.11 0 80-35.89 80-80s-35.89-80-80-80-80 35.89-80 80 35.89 80 80 80zm352-128H304c-8.84 0-16 7.16-16 16v144H64V80c0-8.84-7.16-16-16-16H16C7.16 64 0 71.16 0 80v352c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16v-48h512v48c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16V240c0-61.86-50.14-112-112-112z"
            fill="currentColor"
          />
        </svg>
        3
      </span>
    </div>
    <div>
      <span
        class="property-card_bedrooms"
        data-testid="p_bedrooms-id"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-bath fa-w-16 "
          data-icon="bath"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M32,384a95.4,95.4,0,0,0,32,71.09V496a16,16,0,0,0,16,16h32a16,16,0,0,0,16-16V480H384v16a16,16,0,0,0,16,16h32a16,16,0,0,0,16-16V455.09A95.4,95.4,0,0,0,480,384V336H32ZM496,256H80V69.25a21.26,21.26,0,0,1,36.28-15l19.27,19.26c-13.13,29.88-7.61,59.11,8.62,79.73l-.17.17A16,16,0,0,0,144,176l11.31,11.31a16,16,0,0,0,22.63,0L283.31,81.94a16,16,0,0,0,0-22.63L272,48a16,16,0,0,0-22.62,0l-.17.17c-20.62-16.23-49.83-21.75-79.73-8.62L150.22,20.28A69.25,69.25,0,0,0,32,69.25V256H16A16,16,0,0,0,0,272v16a16,16,0,0,0,16,16H496a16,16,0,0,0,16-16V272A16,16,0,0,0,496,256Z"
            fill="currentColor"
          />
        </svg>
        1
      </span>
    </div>
    <div>
      <span
        class="property-card_price"
        data-testid="p_price-id"
      >
        450000
      </span>
    </div>
    <div>
      <span
        class="property-card_city"
        data-testid="p_city-id"
      >
        Brighton
      </span>
    </div>
    <div>
      <a
        href="mailto:no@no.email.com"
      >
        <span
          class="property-card_email"
          data-testid="p_email-id"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-envelope fa-w-16 "
            data-icon="envelope"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
              fill="currentColor"
            />
          </svg>
          Email
        </span>
      </a>
    </div>
    <button
      class="save-button"
      href="#"
      type="button"
    >
      <svg
        class="save-sign"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M32 448c0 17.7 14.3 32 32 32h384c17.7 0 32-14.3 32-32V160H32v288zm160-212c0-6.6 5.4-12 12-12h104c6.6 0 12 5.4 12 12v8c0 6.6-5.4 12-12 12H204c-6.6 0-12-5.4-12-12v-8zM480 32H32C14.3 32 0 46.3 0 64v48c0 8.8 7.2 16 16 16h480c8.8 0 16-7.2 16-16V64c0-17.7-14.3-32-32-32z"
        />
      </svg>
      Save
      <svg
        class="save-sign"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M433.941 129.941l-83.882-83.882A48 48 0 0 0 316.118 32H48C21.49 32 0 53.49 0 80v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V163.882a48 48 0 0 0-14.059-33.941zM224 416c-35.346 0-64-28.654-64-64 0-35.346 28.654-64 64-64s64 28.654 64 64c0 35.346-28.654 64-64 64zm96-304.52V212c0 6.627-5.373 12-12 12H76c-6.627 0-12-5.373-12-12V108c0-6.627 5.373-12 12-12h228.52c3.183 0 6.235 1.264 8.485 3.515l3.48 3.48A11.996 11.996 0 0 1 320 111.48z"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;
